core/src/cost_model.rs

```diff
@@ -0,0 +1,466 @@
+//! `cost_model` aiming to limit the size of broadcasting sets, and reducing the number
+//! of un-parallelizeble transactions (eg, transactions as same writable key sets).
+//! By doing so to improve leader performance.
+
+use crate::cost_tracker::CostTracker;
+use log::*;
+use solana_sdk::{
+    bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable, clock::Slot, message::Message,
+    pubkey::Pubkey, system_program, transaction::Transaction,
+};
+use std::{collections::HashMap, str::FromStr};
+
+// from mainnet-beta data, taking `vote program` as 1 COST_UNIT to load and execute
+// amount all type programs, the costs are:
```

Comment:
it would be useful to have the hardware that produced this cost.

Reply:
That'd be a solid approach. updated issue page (https://github.com/solana-labs/solana/issues/16641) regarding instruction cost table, will get pipe work done in this PR, then create a PR to implement have `replay stage` to feedback realtime value.